### PR TITLE
feat: Live Updates REST endpoint support

### PR DESCRIPTION
**Pull request by YvesRijckaert**
- Add support for Live Updates REST endpoint
- Rename variables in GraphQL and LiveUpdates class to make it more descriptive

### CODE REVIEW

File: src/rest/entries.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const mergedObj = { ...dataFromPreviewApp };
+
+  for (const field of fields) {
+    const name = field.apiName ?? field.name;
+
+    if (isPrimitiveField(field) || field.type === 'RichText') {
+      updatePrimitiveField(dataFromPreviewApp.fields, updateFromEntryEditor, name, locale);
+      mergedObj.fields[name] = dataFromPreviewApp.fields[name];
+    } else if (field.type === 'Link') {
+      // update single ref field
```
Comment (aodhagan-cf) [inline src/rest/entries.ts:22]: Super nit picky but maybe have these as TODOs

File: src/utils.ts
Comment (aodhagan-cf) [inline src/utils.ts:1]: I'm fine with this for now, just worried that this util file will grow, maybe in the future we can split this up a little bit.

```
import { isPrimitiveField, updatePrimitiveField } from './helpers/field'
import { generateUID } from './helpers/uuid'
import { debounce } from './helpers/util'
```

as a suggestion

File: src/rest/entries.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  locale: string
+): Entity & { sys: SysProps } {
+  if (dataFromPreviewApp.fields) {
+    const updatedFields = { ...dataFromPreviewApp.fields };
+
+    for (const [fieldName, field] of Object.entries(dataFromPreviewApp.fields)) {
+      if (field.sys && field.sys.id === updateFromEntryEditor.sys.id) {
+        updatedFields[fieldName] = mergeFields(fields, field, updateFromEntryEditor, locale);
+        return { ...dataFromPreviewApp, fields: updatedFields };
+      } else if (typeof field === 'object') {
```
Comment (aodhagan-cf) [inline src/rest/entries.ts:44]: Is there any other way to know whether we should call this `updateNestedRef`? just a little confused about what is going on here.

Comment (YvesRijckaert) [reply src/rest/entries.ts:44]: It's to update the fields inside of references, if they have been resolved. It's recursive and will try to find the id of the reference and compare it to the id of the update that is coming from the webapp and then updating it

Comment (chrishelgert) [reply src/rest/entries.ts:44]: I'm probably going to change this a little bit anyways, as I kind of need it also for the asset updates - sorry!

**Pull request merged**